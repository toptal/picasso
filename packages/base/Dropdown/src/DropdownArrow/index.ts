import type { OmitInternalProps } from '@toptal/picasso-shared'

import type { Props } from './DropdownArrow'

export { default as DropdownArrow } from './DropdownArrow'
export type DropdownArrowProps = OmitInternalProps<Props>
