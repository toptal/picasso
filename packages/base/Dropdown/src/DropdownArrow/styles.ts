import { createStyles } from '@material-ui/core/styles'

export default () =>
  createStyles({
    root: {
      width: 0,
      height: 0,
      borderStyle: 'solid',
      borderColor: 'currentColor transparent transparent transparent',
      marginLeft: '0.7em',
    },
    small: {
      borderWidth: '0.25rem 0.25rem 0 0.25rem',
    },
    medium: {
      borderWidth: '0.375rem 0.3125rem 0 0.375rem',
    },
  })
