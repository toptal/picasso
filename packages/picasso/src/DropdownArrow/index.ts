import type { OmitInternalProps } from '@toptal/picasso-shared'

import type { Props } from './DropdownArrow'

export { default } from './DropdownArrow'
export type DropdownArrowProps = OmitInternalProps<Props>
