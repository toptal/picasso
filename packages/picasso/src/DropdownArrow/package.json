{
  "name": "@toptal/picasso-dropdown-arrow",
  "version": "1.0.0",
  "main": "index.js",
  "dependencies": {}
}