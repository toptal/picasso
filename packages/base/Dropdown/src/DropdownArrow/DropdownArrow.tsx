import type { HTMLAttributes } from 'react'
import React, { forwardRef } from 'react'
import type { Theme } from '@material-ui/core/styles'
import { makeStyles } from '@material-ui/core/styles'
import cx from 'classnames'
import type { BaseProps, SizeType } from '@toptal/picasso-shared'

import styles from './styles'

export interface Props extends BaseProps, HTMLAttributes<HTMLSpanElement> {
  /** A Dropdown.Arrow can have different sizes */
  size?: SizeType<'small' | 'medium'>
}

const useStyles = makeStyles<Theme>(styles, {
  name: 'PicassoDropdownArrow',
})

export const DropdownArrow = forwardRef<HTMLSpanElement, Props>(
  function DropdownArrow(props, ref) {
    const { className, style, size = 'medium', ...rest } = props

    const classes = useStyles()

    return (
      <span
        {...rest}
        ref={ref}
        className={cx(classes.root, className, classes[size])}
        style={style}
      />
    )
  }
)

DropdownArrow.displayName = 'DropdownArrow'

DropdownArrow.defaultProps = {
  size: 'medium',
}

export default DropdownArrow
