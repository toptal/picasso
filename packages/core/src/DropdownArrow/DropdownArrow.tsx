import React, { forwardRef, HTMLAttributes } from 'react'
import { withStyles } from '@material-ui/core/styles'
import cx from 'classnames'
import { StandardProps, SizeType } from '@toptal/picasso-shared'

import styles from './styles'

export interface Props extends StandardProps, HTMLAttributes<HTMLSpanElement> {
  /** A Dropdown.Arrow can have different sizes */
  size?: SizeType<'small' | 'medium'>
}

export const DropdownArrow = forwardRef<HTMLSpanElement, Props>(
  function DropdownArrow({ classes, className, style, size, ...rest }, ref) {
    return (
      <span
        // eslint-disable-next-line react/jsx-props-no-spreading
        {...rest}
        ref={ref}
        className={cx(classes.root, className, classes[size!])}
        style={style}
      />
    )
  }
)

DropdownArrow.displayName = 'DropdownArrow'

DropdownArrow.defaultProps = {
  size: 'medium'
}

export default withStyles(styles)(DropdownArrow)
