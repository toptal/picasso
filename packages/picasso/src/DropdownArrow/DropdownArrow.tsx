import React, { forwardRef, HTMLAttributes } from 'react'
import { makeStyles, Theme } from '@material-ui/core/styles'
import cx from 'classnames'
import { BaseProps, SizeType } from '@toptal/picasso-shared'

import styles from './styles'

export interface Props extends BaseProps, HTMLAttributes<HTMLSpanElement> {
  /** A Dropdown.Arrow can have different sizes */
  size?: SizeType<'small' | 'medium'>
}

const useStyles = makeStyles<Theme>(styles, {
  name: 'PicassoDropdownArrow',
})

export const DropdownArrow = forwardRef<HTMLSpanElement, Props>(
  function DropdownArrow(props, ref) {
    const { className, style, size = 'medium', ...rest } = props

    const classes = useStyles()

    return (
      <span
        {...rest}
        ref={ref}
        className={cx(classes.root, className, classes[size])}
        style={style}
      />
    )
  }
)

DropdownArrow.displayName = 'DropdownArrow'

DropdownArrow.defaultProps = {
  size: 'medium',
}

export default DropdownArrow
