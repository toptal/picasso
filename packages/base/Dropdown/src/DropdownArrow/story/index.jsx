import { DropdownArrow } from '../DropdownArrow'
import PicassoBook from '~/.storybook/components/PicassoBook'

const componentDocs = PicassoBook.createComponentDocs(
  DropdownArrow,
  'Dropdown.Arrow'
)

export default { componentDocs }
