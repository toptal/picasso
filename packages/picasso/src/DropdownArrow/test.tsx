import React from 'react'
import { render, cleanup } from '@testing-library/react'

import DropdownArrow from './index'

afterEach(cleanup)

describe('DropdownArrow', () => {
  test('default render', () => {
    const { container } = render(<DropdownArrow />)

    expect(container).toMatchSnapshot()
  })
})
