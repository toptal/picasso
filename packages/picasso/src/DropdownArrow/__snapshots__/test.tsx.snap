// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownArrow renders 1`] = `
<div>
  <span
    class="DropdownArrow-root DropdownArrow-medium"
  />
</div>
`;
